content/en/blog/_posts/2021-07-16-read-write-once-pod-access-mode-alpha.md

```diff
@@ -0,0 +1,267 @@
+---
+layout: blog
+title: "Introducing Single Pod Access Mode for PersistentVolumes"
+date: 2021-07-16
+slug: read-write-once-pod-access-mode-alpha
+---
+
+**Author:** Chris Henzie (Google)
+
+Kubernetes 1.22 introduces a new ReadWriteOncePod access mode for PersistentVolumes and PersistentVolumeClaims that allows users to restrict volume access to a single pod in the cluster.
+
+## What are access modes and why are they important?
+
+When using storage, there are different ways we can model how it is consumed.
+
+For example, a storage system like a network file share can have many users all reading and writing data simultaneously.
+In other cases maybe everyone is allowed to read data but not write it.
+For highly sensitive data, maybe only one user is allowed to read and write data but nobody else.
+
+In the world of Kubernetes, [access modes](https://kubernetes.io/docs/concepts/storage/persistent-volumes/#access-modes) are the way we model how storage is consumed.
+These access modes are a part of the spec for PersistentVolumes (PVs) and PersistentVolumeClaims (PVCs).
+
+```yaml
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+  name: shared-cache
+spec:
+  accessModes:
+  - ReadWriteMany # Allow many pods to access shared-cache simultaneously.
+  resources:
+    requests:
+      storage: 1Gi
+  storageClassName: ""
+```
+
+In Kubernetes, we currently have three access modes for PVs and PVCs:
+
+- ReadWriteOnce -- the volume can be mounted as read-write by a single node
+- ReadOnlyMany -- the volume can be mounted read-only by many nodes
+- ReadWriteMany -- the volume can be mounted as read-write by many nodes
+
+These access modes are enforced by Kubernetes components like the `kube-controller-manager` and `kubelet` to ensure only certain pods are allowed to access a given PersistentVolume.
+
+## What is this new access mode and how does it work?
+
+In Kubernetes v1.22, a fourth access mode for PVs and PVCs is introduced for CSI volumes:
+
+- ReadWriteOncePod -- the volume can be mounted as read-write by a single pod
+
+When a user creates a pod with a PVC that has the ReadWriteOncePod access mode, Kubernetes will ensure that pod is the only pod in the cluster that can read and write to it.
```

Comment:
I couldn't work out from the existing wording whether ReadWriteOncePod blocks another Pod having just read access to that nonvolatile storage.
```suggestion
If you create a pod with a PVC that uses the ReadWriteOncePod access mode, Kubernetes ensures that new pod is the only pod across your whole the cluster that can read that PVC or write to it.
```

Reply:
Thank you for the suggestion. The tricky thing about this wording is "new", it implies a new pod will usurp control over the PVC if it's already in-use by another pod. I kept the sentence but removed the word "new", does it make sense without it?